src/test/java/com/exasol/adapter/dialects/snowflake/SnowflakeVirtualSchemaIntegrationTestSetup.java

```
@@ -0,0 +1,252 @@
+    }
+
+    public Statement getSnowflakeStatement() {
+        return this.snowflakeStatement;
+    }
+
+    public Statement getExasolStatement() {
+        return this.exasolStatement;
+    }
+
+    public ExasolContainer<? extends ExasolContainer<?>> getExasolContainer() {
+        return this.exasolContainer;
+    }
+
+    public VirtualSchema createVirtualSchema(final String forSnowflakeSchema,
+                                             final Map<String, String> additionalProperties) {
+        // TODO add localstack support + cleanup
+        final Map<String, String> properties = new HashMap<>(Map.of("CATALOG_NAME", databaseName, //
+                "SCHEMA_NAME", forSnowflakeSchema)); //
+        properties.putAll(additionalProperties);
+        return this.exasolFactory
+                .createVirtualSchemaBuilder("SNOWFLAKE_VIRTUAL_SCHEMA_" + (this.virtualSchemaCounter++))
+                .adapterScript(this.adapterScript).connectionDefinition(this.connectionDefinition)
+                .properties(properties).build();
+    }
+
+    public ExasolObjectFactory getExasolFactory() {
+        return this.exasolFactory;
+    }
+
+    @Override
+    public void close() {
+        try {
+            this.exasolStatement.close();
+            this.exasolConnection.close();
+            this.snowflakeStatement.close();
+            this.snowflakeConnection.close();
+            this.exasolContainer.stop();
+            // TODO add localstack support + cleanup
+            // this.snowflakeContainer.stop();
+        } catch (final SQLException exception) {
+            throw new IllegalStateException("Failed to stop test setup.", exception);
+        }
+    }
+
+    private String getTestHostIpFromInsideExasol() {
+        final Map<String, ContainerNetwork> networks = this.exasolContainer.getContainerInfo().getNetworkSettings()
+                .getNetworks();
+        if (networks.size() == 0) {
+            return null;
```

Throw an exception here instead of returning null.

---

Any suggestions? Also, there's a assertion after this function call to check it's not null.

---

Ok, since there's an assert, we shouldn't bother, as discussed.